realm/realm-library/src/main/cpp/io_realm_internal_Table.cpp

```diff
@@ -269,9 +269,11 @@ JNIEXPORT jint JNICALL Java_io_realm_internal_Table_nativeGetColumnType(JNIEnv*,
     ColKey column_key (columnKey);
     TableRef table = TBL_REF(nativeTableRefPtr);
     jint column_type = table->get_column_type(column_key);
-    if (column_type != type_LinkList &&  table->is_list(column_key)) {
+    if (column_type != type_LinkList && column_key.is_list()) {
         // add the offset so it can be mapped correctly in Java (RealmFieldType#fromNativeValue)
-        column_type += 128;
+        column_type += 128;     // PropertyType::Array
```

Comment:
Don't have a more type safe reference to these values? I.e you can static_cast the PropertyType?